pkg/apply/applier_apply_options.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2021 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package apply
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/cli-utils/pkg/apply/event"
+	"sigs.k8s.io/cli-utils/pkg/common"
+	"sigs.k8s.io/cli-utils/pkg/inventory"
+)
+
+const defaultPollInterval = 2 * time.Second
+
+type Options struct {
```

Comment:
Given that we're changing how to build this, we might want to rename it to `apply.ApplierOptions` too, to be consistent with the `DestroyerOptions`. I think the applier options came first before the destroyer was added.

Also, do we want to make this a private struct so it has to be constructed with the builder?

Reply:
Makes sense, renamed and made private.

Reply:
Hm, actually, if we make it private it will be difficult to use mocks with it. Mocking library will not be able to refer to the type.